javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -674,6 +674,44 @@ default int length() {
     @Override
     Map<K, V> replace(Tuple2<K, V> currentElement, Tuple2<K, V> newElement);
 
+    /**
+     * Replaces the entry for the specified key only if it is currently mapped to some value.
+
+     * @param key the key of the element to be substituted.
+     * @param value the new value to be associated with the key
+     * @return a new map containing key mapped to value if key was contained before. The old map otherwise.
+     */
+    default Map<K, V> replaceCurrentValue(K key, V value) {
+        Objects.requireNonNull(key, "key is null");
+        return this.get(key)
+                   .map(currentValue -> this.replace(Tuple.of(key, currentValue), Tuple.of(key, value)))
+                   .getOrElse(this);
+    }
+
+    /**
+     * Replaces the entry for the specified key only if currently mapped to the specified value.
+     * @param key the key of the element to be substituted.
+     * @param oldValue the expected current value that the key is currently mapped to
+     * @param newValue the new value to be associated with the key
+     * @return a new map containing key mapped to newValue if key was contained before and oldValue matched. The old map otherwise.
+     */
+    default Map<K, V> replace(K key, V oldValue, V newValue) {
+        Objects.requireNonNull(key, "key is null");
+        return this.get(key)
+                   .filter(current -> current.equals(oldValue))
+                   .map(currentValue -> this.replace(Tuple.of(key, currentValue), Tuple.of(key, newValue)))
+                   .getOrElse(this);
```

Comment:
For me just `return map.containsKey(key) ? put(map, Tuple(key, value)) :  map;` works.

`return map.containsKey(key) ? map.put(key, value) :  map;` gives a compilation error.

`Error:(244, 46) java: incompatible types: bad type in conditional expression
    javaslang.collection.Map<K,V> cannot be converted to M`

I would rather use the first version and avoid the cast to M. Is that OK?